File: src/doc_analyse/ingestion/pipeline.py
from __future__ import annotations

from pathlib import Path
from typing import Optional, Union

from doc_analyse.ingestion.chunking import DEFAULT_CHUNK_OVERLAP, DEFAULT_CHUNK_SIZE, TextChunker
from doc_analyse.ingestion.converters import ConverterRegistry, convert_document
from doc_analyse.ingestion.models import IngestedDocument


def ingest_document(
    path: Union[str, Path],
    *,
    registry: Optional[ConverterRegistry] = None,
    chunker: Optional[TextChunker] = None,
    chunk_size: int = DEFAULT_CHUNK_SIZE,
    chunk_overlap: int = DEFAULT_CHUNK_OVERLAP,
) -> IngestedDocument:
    """Convert one uploaded file and split the normalized text into chunks."""
    document = convert_document(path, registry=registry)
    resolved_chunker = chunker or TextChunker(
        chunk_size=chunk_size,
        chunk_overlap=chunk_overlap,
    )

    return IngestedDocument(
        document=document,
        chunks=resolved_chunker.chunk(document),
    )

File: src/ingestion/pipeline.ts
/**
 * Ingestion pipeline.
 * Ported from doc_analyse/ingestion/pipeline.py
 */

import { resolve } from "path";
import { createIngestedDocument, IngestedDocument } from "./models.js";
import { TextChunker, DEFAULT_CHUNK_SIZE, DEFAULT_CHUNK_OVERLAP } from "./chunking.js";
import { ConverterRegistry, convert_document } from "./converters.js";

export function ingest_document(
  path: string,
  {
    registry,
    chunker,
    chunkSize = DEFAULT_CHUNK_SIZE,
    chunkOverlap = DEFAULT_CHUNK_OVERLAP,
  }: {
    registry?: ConverterRegistry;
    chunker?: TextChunker;
    chunkSize?: number;
    chunkOverlap?: number;
  } = {},
): IngestedDocument {
  const resolvedPath = resolve(path);
  const document = convert_document(resolvedPath, registry);

  const resolvedChunker =
    chunker ??
    new TextChunker({ chunkSize, chunkOverlap });

  return createIngestedDocument(document, resolvedChunker.chunk(document));
}
